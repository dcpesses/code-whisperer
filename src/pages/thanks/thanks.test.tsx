import {render} from '@testing-library/react';
import Thanks from './index';

interface LinkProps {
  className: string,
  to: string,
  children: React.ReactNode
}

vi.mock('react-router-dom', () => {
  const reactRouterDom = vi.importActual('react-router-dom');
  return {
    ...reactRouterDom,
    Link: ({className, to, children}: LinkProps) => (
      <div className={className} data-to={to}>
        {children}
      </div>
    )
  };
});

describe('Thanks', () => {
  test('Should render as expected', () => {
    const {container} = render(<Thanks />);
    expect(container).toMatchSnapshot();
  });
});
