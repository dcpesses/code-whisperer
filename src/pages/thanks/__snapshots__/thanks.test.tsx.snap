// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Thanks > Should render as expected 1`] = `
<div>
  <div
    class="thanks container-fluid d-flex justify-content-center align-items-center vw-100 vh-100"
  >
    <main
      class="row text-center d-flex align-items-center"
    >
      <h1
        class="pb-4"
      >
        Thanks!
      </h1>
      <p>
        We've received your submission.
      </p>
      <div
        class="col-6 mx-auto"
      >
        <div
          class="btn btn-outline-primary mt-5"
          data-to="/"
        >
          Back
        </div>
      </div>
    </main>
  </div>
</div>
`;
